packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -472,8 +472,17 @@ class _WindowsUtils extends OperatingSystemUtils {
     required super.processManager,
   }) : super._private();
 
+  HostPlatform? _hostPlatform;
+
   @override
-  HostPlatform hostPlatform = HostPlatform.windows_x64;
+  HostPlatform get hostPlatform {
```

Comment:
Is this PR about host, or target, or both? The PR description currently sounds like it's only about the target, but this is changing host.

Reply:
Initially, I wanted to upstream everything to prepare arm64 support.
Basically, the option --target-platform could use this new Host Platform to derive automatically default target.

This code won't be used right now though, as the --target-platform flag has a windows-x64 value. If you think it's not worth it (it will have to be committed at some point), I can drop it.

Reply:
For now, only native workflows have been tried (x64 -> x64, arm64 -> arm64). Doing cross compilation implies work with gen_snapshot, which I didn't start to look into.

Reply:
I would strongly recommend a design document for any tool changes; it doesn't really make sense to decide what the tool commands, output, etc. should look like without a discussion of whether cross-compilation will be supported. If same-arch builds are the only option then it doesn't seem like we even *need* a target flag.

Reply:
I'll focus on getting flutter engine built and uploaded for windows-arm64 for now. Then, we can reopen that discussion.